dhall/src/Dhall/Core.hs

```diff
@@ -336,7 +336,7 @@ instance Pretty Import where
     Zero indices are omitted when pretty-printing `Var`s and non-zero indices
     appear as a numeric suffix.
 -}
-data Var = V Text !Integer
+data Var = V Text !Int
```

Comment:
Maybe go all the way and `{-# UNPACK #-}`?

Reply:
Actually, I think GHC already unpacks these strict fields now.

Reply:
Yeah, I believe GHC will automatically unbox strict fields if they are no larger than a pointer.

Note that I haven't actually benchmarked this; I'm only forward-porting small chunks of the work mentioned in #1039